airship/kubeadm-airship-gap-analysis.md

```diff
@@ -1,14 +1,53 @@
 # Introduction
 The purpose of this document is for studying the limitations of kubeadm in accomplishing the tasks done by airship components.
 Each task is mapped to a jira issue and focuses on some specific topic as outlined during a face to face meeting. The original meeting points can found in Gaps section here: [Airship_F2f_Notes](https://etherpad.openstack.org/p/Airship_F2f_Notes)
- 
 
 # Load Balancers
 
 Jira Issues: 
 - [metallb](https://airship.atlassian.net/browse/AIR-5)
 - [keepalived](https://airship.atlassian.net/browse/AIR-140)
 
+## Introduction
+
+There are multiple alternatives for HA setup. Time wise, Some are setup before a kubernetes cluster is created, while other need the target cluster in place. Based on the following criteria, an assessment was done to determine their usability with kubeadm.
+
+1. Can we have static information that can be used in certificates
+2. Can we have all required information before creating the cluster
+3. Can new control plane nodes join dynamically
+4. Can clusters survive control plane node removal
+
+## HA options
+
+### Keepalived
+- Each node will have its own ip and one common IP among them which is static.
+- The common IP (VIP) and each VM’s IP can be used in certificates
+- Each joining node can join the “keepalived group” much like the original nodes then configures kubeadm to use the VIP as a cluster IP.
+- Node removal does not cause total failure of the cluster, but downtime if the node had the VIP
+
```

Comment:
If Node and VM are referring to same entity from lines 23-26 then it would be good to use only one notion to make the text consistent

Reply:
done